### ISSUE TITLE
Associate CSB JSON documents with a particular schema version

### ISSUE THREAD

**Issue by selimnairb** [CCOMJHC/csbschema#14]
There should be a way to assert that a particular JSON-encoded CSB document should be validated against a particular CSB schema version. The convention within JSON Schema usage appears to be to include a "$schema" property whose value is the URI/URL of a schema file (i.e. the value of the "$id" property defined in that JSON Schema file). 

For example, a [document](https://github.com/CCOMJHC/csbschema/blob/main/docs/IHO/b12_v3_1_0_example.json) that should be validated against version [3.1.0-2023-08](https://github.com/CCOMJHC/csbschema/blob/main/csbschema/data/CSB-schema-3_1_0-2023-08.json) of the CSB schema could include the following property at its top level:
```
{
  "$schema": "https://raw.githubusercontent.com/CCOMJHC/csbschema/main/csbschema/data/CSB-schema-3_1_0-2023-08.json"
  ...
}
```

`$schema` would be an optional, but highly recommended property to be included in JSON-encoded CSB documents. As part of work to address this issue, `csbschema` validation code would be updated to autodetect the schema document to use when validating a particular document. This would allow for more flexibility in processing JSON-encoded CSB documents, especially where many files from multiple observers/trusted nodes need to be processed in bulk.

**Comment by cclint** [comment]
@selimnairb is the repo ready for transfer or are you still doing some house cleaning?

**Comment by selimnairb** [comment]
> [@selimnairb](https://github.com/selimnairb) is the repo ready for transfer or are you still doing some house cleaning?

Yes, it's ready to go now. Sorry for not responding sooner, I wanted to do a final sweep for references in comments and documentation, which I have now done. Thanks!
